### PR TITLE
Add laravel sample app

### PR DESCRIPTION
**Pull request by PierrickVoulet**
This is based on the [existing one](https://github.com/googleads/googleads-php-lib/tree/master/examples/AdWords/LaravelSampleApp) developed for the AdWords API client library for PHP.

The use cases that are used here changed though:

* Show report results using GAS.SearchStream
* Pause a campaign using CampaignService.MutateCampaigns and then retrieve its information using GAS.Search

You are welcome to review all the files but most of them have been generated by Laravel so it would not be the best use of your time. My advice: focus on the files that contains the `Copyright 2020 Google LLC` license header.

### CODE REVIEW

File: examples/LaravelSampleApp/app/Http/Controllers/GoogleAdsApiController.php [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+
+class GoogleAdsApiController extends Controller
+{
+    private static $REPORT_TYPE_TO_DEFAULT_SELECTED_FIELDS = [
+        'campaign' => ['campaign.id', 'campaign.name', 'campaign.status'],
+        'customer' => ['customer.id']
+    ];
+
+    /**
+     * Controls a POST or GET request submitted in the contect of the "Show Report" form.
```
Comment (fiboknacky) [inline examples/LaravelSampleApp/app/Http/Controllers/GoogleAdsApiController.php:42]: contect?

File: examples/LaravelSampleApp/app/Http/Controllers/GoogleAdsApiController.php [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+            // Searches the results using streaming.
+            /** @var GoogleAdsServerStreamDecorator $stream */
+            $stream = $googleAdsClient->getGoogleAdsServiceClient()->searchStream(
+                $clientCustomerId,
+                $query
+            );
+
+            // Fetches all the results.
+            $results = [];
+            foreach ($stream->readAll() as $response) {
```
Comment (fiboknacky) [inline examples/LaravelSampleApp/app/Http/Controllers/GoogleAdsApiController.php:96]: We cannot use the interceptor to hide a nested loop here?

File: examples/LaravelSampleApp/app/Http/Controllers/GoogleAdsApiController.php [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+                /** @var SearchGoogleAdsStreamResponse $response */
+                foreach ($response->getResults() as $googleAdsRow) {
+                    /** @var GoogleAdsRow $googleAdsRow */
+                    // Converts each result as a POPO using JSON.
```
Comment (fiboknacky) [inline examples/LaravelSampleApp/app/Http/Controllers/GoogleAdsApiController.php:100]: Could you spell it out? I believe POPO is not popular as POJO.

File: examples/LaravelSampleApp/app/Http/Controllers/GoogleAdsApiController.php [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+                    $results[] = json_decode($googleAdsRow->serializeToJsonString(), true);
+                }
+            }
+            $collection = collect($results);
+
+            // Updates the session with all the information that is necessary to process any
+            // following GET requests (report result pages).
+            $request->session()->put('selectedFields', $selectedFields);
+            $request->session()->put('entriesPerPage', $entriesPerPage);
+            $request->session()->put('collection', collect($collection));
```
Comment (fiboknacky) [inline examples/LaravelSampleApp/app/Http/Controllers/GoogleAdsApiController.php:110]: Why do we need two collect here?
collect(collect($results))?

File: examples/LaravelSampleApp/app/Http/Controllers/GoogleAdsApiController.php (hunk 2 above)
Comment (PierrickVoulet) [reply examples/LaravelSampleApp/app/Http/Controllers/GoogleAdsApiController.php:96]: This is not released in 3.1.0. I will have to wait for 3.2.0 to make the change.

File: examples/LaravelSampleApp/app/Http/Controllers/GoogleAdsApiController.php (hunk 4 above)
Comment (PierrickVoulet) [reply examples/LaravelSampleApp/app/Http/Controllers/GoogleAdsApiController.php:110]: Good catch!

**Comment by fiboknacky** [comment]
Pierrick, how about first blindly copying what you need from laravel/framework first? This way, at least, we still merge what's always functioning.
Then, in the next PR, you can modify what you did here.

**Comment by PierrickVoulet** [comment]
@AnashOommen @fiboknacky I opened another PR #328 to help the review. Once #328 is merged, it should be much easier to review this PR.

**Comment by PierrickVoulet** [comment]
@AnashOommen @fiboknacky it is now ready for review (and should be much easier to review).

File: examples/LaravelSampleApp/app/Http/Controllers/GoogleAdsApiController.php [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+
+        // Builds the GAQL query to retrieve more information about the now paused campaign.
+        $query = sprintf(
+            "SELECT campaign.id, campaign.name, campaign.status FROM campaign " .
+            "WHERE campaign.resource_name = '%s' LIMIT 1",
+            $campaignResourceName
+        );
+
+        // Searches the result using pagination.
+        $response = $googleAdsClient->getGoogleAdsServiceClient()->search(
```
[the remark this answers is not in the document]
Comment (PierrickVoulet) [reply examples/LaravelSampleApp/app/Http/Controllers/GoogleAdsApiController.php]: Good point.

**Comment by PierrickVoulet** [comment]
> Left some small comments first. I'll try the app myself soon today or tomorrow.

SG, I fixed your comments and rewinded the review request.

**Comment by fiboknacky** [comment]
I'm facing the SSL eof error, which is not related to this project. But it does prevent me from seeing the result page, so let's wait a few days until I can resolve this issue. 
(If you have a clue, please let me know)

BTW, could you include an outstanding warning message in the main webpage that the user needs to have grpc installed?
I tried in a separate environment but failed. Although the Laravel's stack-trace page is not bad, it'd be friendly to beginners to know this in advance.

**Comment by fiboknacky** [comment]
> I'm facing the SSL eof error, which is not related to this project. But it does prevent me from seeing the result page, so let's wait a few days until I can resolve this issue.
> (If you have a clue, please let me know)
> 
> BTW, could you include an outstanding warning message in the main webpage that the user needs to have grpc installed?
> I tried in a separate environment but failed. Although the Laravel's stack-trace page is not bad, it'd be friendly to beginners to know this in advance.

I found the fix. It seems openssl 1.1.1e contains [a bug](https://help.whmcs.com/m/troubleshooting/l/1230985-troubleshooting-curl-error-56-openssl-ssl_read-unexpected-eof-while-reading). I need to upgrade it to 1.1.1f.

**Comment by PierrickVoulet** [comment]
> Pending the request to add a message for those who don't use grpc.

Good point, I added a warning in the header.

Once the transport property will be released #331 , I think we should programmatically force the transport to gRPC so that there is no ambiguity in case of error at runtime.

**Pull request merged**